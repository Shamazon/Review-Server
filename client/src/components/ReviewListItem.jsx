import React from 'react';
import UserInfo from './UserInfo.jsx';
import ReviewTitle from './ReviewTitle.jsx';
import ReviewBody from './ReviewBody.jsx'
import style from '../../dist/style/style.css'

var ReviewListItem = (props) => {
    return (
        <article className={style.mainReviewContainer}>
            <UserInfo />
            <ReviewTitle review={props.review} />
            <div>
                <span className={style.gray}>Color: seafoam</span>
                <span className={style.seperator}>|</span>
                <span className={style.verified}>Verified Purchase</span>
            </div>
            
            <ReviewBody body={props.review.body} />
            
            <div className={`${style.helpful} ${style.gray}`}>
            <a className={style.underline}>Comment</a><span className={style.seperator}>|</span>
                <span>{props.review.helpful} people found this review helpful. Was it helpful to you?</span> 
                <button>Yes</button><button>No</button><a className={style.aSecondary}>Report Abuse</a>
            </div>
        </article>
    )
}

export default ReviewListItem;