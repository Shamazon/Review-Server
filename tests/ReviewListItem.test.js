import React from 'react';
import ReviewListItem from '../client/src/components/ReviewListItem.jsx';
import enzyme from 'enzyme';

describe('ReviewListItem', () => {
  it('should be defined', () => {
    expect(ReviewListItem).toBeDefined();
  });
  it('should render correctly', () => {
    const props = {
      review: {
        body: 'test',
      },
    };
    const tree = enzyme.shallow(<ReviewListItem {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
