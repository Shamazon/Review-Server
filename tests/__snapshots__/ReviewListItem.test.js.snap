// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewListItem should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ReviewListItem
    review={
      Object {
        "body": "test",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <UserInfo />,
        <ReviewTitle
          review={
            Object {
              "body": "test",
            }
          }
        />,
        <div>
          <span
            className="gray"
          >
            Color: seafoam
          </span>
          <span
            className="seperator"
          >
            |
          </span>
          <span
            className="verified"
          >
            Verified Purchase
          </span>
        </div>,
        <ReviewBody
          body="test"
        />,
        <div
          className="helpful gray"
        >
          <a
            className="underline"
          >
            Comment
          </a>
          <span
            className="seperator"
          >
            |
          </span>
          <span>
             people found this review helpful. Was it helpful to you?
          </span>
          <button>
            Yes
          </button>
          <button>
            No
          </button>
          <a
            className="aSecondary"
          >
            Report Abuse
          </a>
        </div>,
      ],
      "className": "mainReviewContainer",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "review": Object {
            "body": "test",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="gray"
            >
              Color: seafoam
            </span>,
            <span
              className="seperator"
            >
              |
            </span>,
            <span
              className="verified"
            >
              Verified Purchase
            </span>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Color: seafoam",
              "className": "gray",
            },
            "ref": null,
            "rendered": "Color: seafoam",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "|",
              "className": "seperator",
            },
            "ref": null,
            "rendered": "|",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Verified Purchase",
              "className": "verified",
            },
            "ref": null,
            "rendered": "Verified Purchase",
            "type": "span",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "body": "test",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <a
              className="underline"
            >
              Comment
            </a>,
            <span
              className="seperator"
            >
              |
            </span>,
            <span>
               people found this review helpful. Was it helpful to you?
            </span>,
            <button>
              Yes
            </button>,
            <button>
              No
            </button>,
            <a
              className="aSecondary"
            >
              Report Abuse
            </a>,
          ],
          "className": "helpful gray",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Comment",
              "className": "underline",
            },
            "ref": null,
            "rendered": "Comment",
            "type": "a",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "|",
              "className": "seperator",
            },
            "ref": null,
            "rendered": "|",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                undefined,
                " people found this review helpful. Was it helpful to you?",
              ],
            },
            "ref": null,
            "rendered": Array [
              undefined,
              " people found this review helpful. Was it helpful to you?",
            ],
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Yes",
            },
            "ref": null,
            "rendered": "Yes",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "No",
            },
            "ref": null,
            "rendered": "No",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Report Abuse",
              "className": "aSecondary",
            },
            "ref": null,
            "rendered": "Report Abuse",
            "type": "a",
          },
        ],
        "type": "div",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <UserInfo />,
          <ReviewTitle
            review={
              Object {
                "body": "test",
              }
            }
          />,
          <div>
            <span
              className="gray"
            >
              Color: seafoam
            </span>
            <span
              className="seperator"
            >
              |
            </span>
            <span
              className="verified"
            >
              Verified Purchase
            </span>
          </div>,
          <ReviewBody
            body="test"
          />,
          <div
            className="helpful gray"
          >
            <a
              className="underline"
            >
              Comment
            </a>
            <span
              className="seperator"
            >
              |
            </span>
            <span>
               people found this review helpful. Was it helpful to you?
            </span>
            <button>
              Yes
            </button>
            <button>
              No
            </button>
            <a
              className="aSecondary"
            >
              Report Abuse
            </a>
          </div>,
        ],
        "className": "mainReviewContainer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "review": Object {
              "body": "test",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="gray"
              >
                Color: seafoam
              </span>,
              <span
                className="seperator"
              >
                |
              </span>,
              <span
                className="verified"
              >
                Verified Purchase
              </span>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Color: seafoam",
                "className": "gray",
              },
              "ref": null,
              "rendered": "Color: seafoam",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "|",
                "className": "seperator",
              },
              "ref": null,
              "rendered": "|",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Verified Purchase",
                "className": "verified",
              },
              "ref": null,
              "rendered": "Verified Purchase",
              "type": "span",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "body": "test",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <a
                className="underline"
              >
                Comment
              </a>,
              <span
                className="seperator"
              >
                |
              </span>,
              <span>
                 people found this review helpful. Was it helpful to you?
              </span>,
              <button>
                Yes
              </button>,
              <button>
                No
              </button>,
              <a
                className="aSecondary"
              >
                Report Abuse
              </a>,
            ],
            "className": "helpful gray",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Comment",
                "className": "underline",
              },
              "ref": null,
              "rendered": "Comment",
              "type": "a",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "|",
                "className": "seperator",
              },
              "ref": null,
              "rendered": "|",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  undefined,
                  " people found this review helpful. Was it helpful to you?",
                ],
              },
              "ref": null,
              "rendered": Array [
                undefined,
                " people found this review helpful. Was it helpful to you?",
              ],
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Yes",
              },
              "ref": null,
              "rendered": "Yes",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "No",
              },
              "ref": null,
              "rendered": "No",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Report Abuse",
                "className": "aSecondary",
              },
              "ref": null,
              "rendered": "Report Abuse",
              "type": "a",
            },
          ],
          "type": "div",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
